Src/FluentAssertions/Execution/ContextDataItems.cs

```diff
@@ -42,10 +42,13 @@ public void Add(ContextDataItems contextDataItems)
 
         public void Add(DataItem item)
         {
-            DataItem existingItem = items.SingleOrDefault(i => i.Key == item.Key);
-            if (existingItem != null)
+            for (int i = 0; i < items.Count; i++)
```

Comment:
🔧 What if you would have `DataItem` implement `IEquatable`? Wouldn't that simplify and improve the speed? You can then use `IndexOf`

Reply:
It might simplify it a bit but I'm not convinced it would affect speed that much. The big thing was eliminating `.Remove` which has to shuffle elements around. I don't think this list actually gets all that big either.

Reply:
Sure. It was just a suggestion.